RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate2D.cc

```diff
@@ -1,801 +1,1498 @@
+            >> theCurrentTemp.entry[iy][jx].offsetx[2] >> theCurrentTemp.entry[iy][jx].offsetx[3]>> theCurrentTemp.entry[iy][jx].offsety[0] >> theCurrentTemp.entry[iy][jx].offsety[1]
+            >> theCurrentTemp.entry[iy][jx].offsety[2] >> theCurrentTemp.entry[iy][jx].offsety[3];
+            
+            if(in_file.fail()) {LOGERROR("SiPixelTemplate2D") << "Error reading file 6, no template load, run # " << theCurrentTemp.entry[iy][jx].runnum << ENDL; return false;}
+            
+            in_file >> theCurrentTemp.entry[iy][jx].clsleny >> theCurrentTemp.entry[iy][jx].clslenx >> theCurrentTemp.entry[iy][jx].mpvvav >> theCurrentTemp.entry[iy][jx].sigmavav
+            >> theCurrentTemp.entry[iy][jx].kappavav >> theCurrentTemp.entry[iy][jx].scalexavg >> theCurrentTemp.entry[iy][jx].scaleyavg >> theCurrentTemp.entry[iy][jx].delyavg >> theCurrentTemp.entry[iy][jx].delysig >> theCurrentTemp.entry[iy][jx].spare[0];
+            
+            if(in_file.fail()) {LOGERROR("SiPixelTemplate2D") << "Error reading file 7, no template load, run # " << theCurrentTemp.entry[iy][jx].runnum << ENDL; return false;}
+            
+            in_file >> theCurrentTemp.entry[iy][jx].scalex[0] >> theCurrentTemp.entry[iy][jx].scalex[1] >> theCurrentTemp.entry[iy][jx].scalex[2] >> theCurrentTemp.entry[iy][jx].scalex[3]
+            >> theCurrentTemp.entry[iy][jx].scaley[0] >> theCurrentTemp.entry[iy][jx].scaley[1] >> theCurrentTemp.entry[iy][jx].scaley[2] >> theCurrentTemp.entry[iy][jx].scaley[3]
+            >> theCurrentTemp.entry[iy][jx].spare[1]  >> theCurrentTemp.entry[iy][jx].spare[2];
+            
+            if(in_file.fail()) {LOGERROR("SiPixelTemplate2D") << "Error reading file 8, no template load, run # " << theCurrentTemp.entry[iy][jx].runnum << ENDL; return false;}
+            
+         }
+         
+      }
+      
+      
+      in_file.close();
+      
+      // Add this template to the store
+      
+      thePixelTemp_.push_back(theCurrentTemp);
+      
+      return true;
+      
+   } else {
+      
+      // If file didn't open, report this
+      
+      LOGERROR("SiPixelTemplate2D") << "Error opening File" << tempfile << ENDL;
+      return false;
+      
+   }
+   
+} // TempInit
+
+
+#ifndef SI_PIXEL_TEMPLATE_STANDALONE
+
+//****************************************************************
+//! This routine initializes the global template structures from an
+//! external file template_summary_zpNNNN where NNNN are four digits
+//! \param dbobject - db storing multiple template calibrations
+//****************************************************************
+bool SiPixelTemplate2D::pushfile(const SiPixel2DTemplateDBObject& dbobject, std::vector< SiPixelTemplateStore2D > & thePixelTemp_)
+{
+   // Add template stored in external dbobject to theTemplateStore
+   
+   // Local variables
+   int i, j, k, l, iy, jx;

```

Comment:
are these really needed in this scope?
I see that ``` i ``` is used in ```for (i=0; i<20; ++i) {```
It should be moved down to the lowest scope needed for the variable

Reply:
This is too low-level so I'd like to discuss this with Morris.  But I made a note in the comments.